src/modules/item/item.controller.ts

```diff
@@ -68,6 +68,32 @@ export default class ItemController {
 		}
 	}
 
+	public async readHandler(
+		request: FastifyRequest<{
+			Params: itemReadInput;
+		}>,
+		reply: FastifyReply,
+	) {
+		try {
+			const id = Number.parseInt(request.params.id);
+
+			if (!(await this.accessService.hasAccessToItem(id, request.user.sub))) {
+				return reply.unauthorized();
+			}
+
+			const item = await this.itemService.getItemByIdWithFolderAndDocsAndBlob(id);
+
+			return reply.code(200).send(item);
```

Comment:
As itemService does not throw an error, calling it with an invalid id will return status 200 and it will try to return undefined in response body.

The only reason this does not happen is the hasAccessToItem call checking if it exists.